Fallback for Complex Dtypes in Inductor

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #97198
* #98869



cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

eellison committed:
Fallback for Complex Dtypes in Inductor

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison commented on torch/_inductor/codegen/cpp.py:
```diff
@@ -2083,7 +2083,9 @@ def _used_by_to(to_node: torch.fx.Node):
                                 sub_graph.erase_node(node)
 
-                    sub_graph.lint()
+                    # throws bound method LoopBody.get_index of <torch._inductor.ir.LoopBody object
```
cc @EikanWang would you take a look at this ? lint() throws because `LoopBody.get_index` is the `call_module` node but `LoopBody.get_index` is not an nn.Module

EikanWang replied:
Oh, sure. Let's check it.

EikanWang replied:
@eellison , may I know if this issue only happens under debug mode?

EikanWang replied:
For debug mode, the graph of `LoopBody` will attach a new `GraphModule` as `owning_module` while the release mode will not - https://github.com/pytorch/pytorch/blob/master/torch/_inductor/ir.py#L4203. The `lint` will check whether the graph has `owning_module` to decide if it needs to check `call_module` - https://github.com/pytorch/pytorch/blob/master/torch/fx/graph.py#L1338. Hence, we could patch the `owning_graph`  of `sub_graph` to `None` to move the `lint` onward. Eventually, I think we should call `get_index` via `call_function` but not `call_module`. I will submit a PR to fix this issue.

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison committed:
Update on "Fallback for Complex Dtypes in Inductor"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

Differential Revision: [D44257054](https://our.internmc.facebook.com/intern/diff/D44257054)

[ghstack-poisoned]

eellison commented:
@pytorchbot merge -f "flakey failure"
